units/src/result.rs

```diff
@@ -9,6 +9,74 @@ use NumOpResult as R;
 use crate::{Amount, FeeRate, SignedAmount, Weight};
 
 /// Result of a mathematical operation on two numeric types.
+///
+/// For some types in this crate overflow can be catastrophic (e.g. `Amount`). In order to prevent
+/// overflow we provide a custom result type that is similar to the normal [`core::result::Result`]
+/// but implements mathematical operatons (e.g. [`core::ops::Add`]) so that math operations can be
+/// chained ergonomically.
```

Comment:
I suggest adding: "This is very similar to how `NaN` works."

I think it'll be much more familiar explained that way.

Reply:
Added.